### PR TITLE
feat(types): use retry-as-promised types for retry options to match documentation

### PR DESCRIPTION
**Pull request by esetnik**
## Pull Request Checklist

- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change
I have corrected the types in the underlying `retry-as-promised` library so they can now be re-exported by sequelize to get the proper type for `retry`.

See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63267 for details.

### CODE REVIEW

**Review by WikiRik — DISMISSED** [review]
There are a few small differences between the current state and the types package but I do think we should prefer the types package over our own type. Thanks for this PR!

File: yarn.lock
Comment (ephys) [inline yarn.lock:1117]: damn these types are old

**Review by ephys — CHANGES_REQUESTED** [review]

File: yarn.lock
Comment (ephys) [reply yarn.lock:1117]: Oh that's very strange, there was a release on npm 3 months ago https://www.npmjs.com/package/retry-as-promised

Comment (ephys) [reply yarn.lock:1117]: Oh nevermind it's just not listed in github releases but it's still getting updated https://github.com/mickhansen/retry-as-promised/blob/master/package.json

Comment (esetnik) [reply yarn.lock:1117]: Actually I was wrong. They stopped managing github releases but have still been releasing source code changes. https://www.npmjs.com/package/retry-as-promised

Comment (ephys) [reply yarn.lock:1117]: By the way this isn't a problem for this PR, our users won't be affected since we only use Options

Comment (WikiRik) [reply yarn.lock:1117]: Yeah, it's a package made by one of the former sequelize maintainers. Might be good if we can get this project under the sequelize organisation to ensure the continuity better.

@mickhansen not sure how often you check this anymore, but what do you think?

Comment (ephys) [reply yarn.lock:1117]: Thank you! :)

We can still merge this, renovate will handle opening the package update PR

File: package.json
Comment (ephys) [inline package.json:50]: We don't pin our production dependencies so tools like `npm audit fix` continue working :)

```suggestion
    "@types/retry-as-promised": "^2.3.4",
```

File: yarn.lock
Comment (fzn0x) [reply yarn.lock:1117]: > Yeah, it's a package made by one of the former sequelize maintainers. Might be good if we can get this project under the sequelize organisation to ensure the continuity better.
> 
> @mickhansen not sure how often you check this anymore, but what do you think?

If this idea is approved, I think it's better that the repository created by other maintainers can also be moved into the organization, do you think that's better?

This is just to eliminate the possibility of the library becoming obsolete in the future if it's under one person, not that I expect that.

**Review by ephys — APPROVED** [review]
🚀🎉

File: yarn.lock
Comment (fzn0x) [reply yarn.lock:1117]: > That would be the idea
> 
> But note that we can always fork if one of our dependencies becomes unmaintained, we did that recently with https://github.com/sequelize/toposort

Ah that's right

**Review by WikiRik — APPROVED** [review]

File: yarn.lock
Comment (mickhansen) [reply yarn.lock:1117]: I don't often check the Sequelize repository, but do try to keep on top of the packages directly under my GitHub user.
Looks like I forgot to actually publish the tags as releases on retry-as-promised.

**Comment by esetnik** [comment]
Note that @mickhansen is going to refactor `retry-as-promised` to typescript so we can hold off on merging this PR. I will update this PR to depend on the new version of `retry-as-promised` with native types once it has been released.

File: yarn.lock
Comment (mickhansen) [reply yarn.lock:1117]: @esetnik https://github.com/mickhansen/retry-as-promised/releases/tag/v7.0.0

**Comment by esetnik** [comment]
Waiting on https://github.com/mickhansen/retry-as-promised/pull/37

**Review by ephys — APPROVED** [review]
If tests are green, we're good to go :)

**Comment by esetnik** [comment]
@ephys will this get a backport release to v6?

**Comment by WikiRik** [comment]
@esetnik we should be able to, can you open a new PR towards the v6 branch?

**Pull request merged**
